metrics/options.go

```diff
@@ -0,0 +1,84 @@
+package metrics
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/coopnorge/go-datadog-lib/v2/errors"
+	"github.com/coopnorge/go-datadog-lib/v2/internal"
+	"github.com/coopnorge/go-logger"
+)
+
+const (
+	defaultEnableMetrics    = true
+	defaultMetricSampleRate = 1
+)
+
+// Option is used to configure the behaviour of the metrics integration.
+type Option func(*options) error
+
+type options struct {
+	errorHandler     errors.ErrorHandler
+	dsdEndpoint      string
+	metricSampleRate float64
+	tags             []string
+}
+
+func resolveOptions(opts []Option) (*options, error) {
+	err := internal.VerifyEnvVarsSet(
+		internal.DatadogDSDEndpoint,
+		internal.DatadogEnvironment,
+		internal.DatadogService,
+		internal.DatadogVersion,
+	)
+	if err != nil {
+		return nil, err
+	}
+	options := &options{
+		errorHandler: func(err error) {
+			logger.WithError(err).Error(err.Error())
+		},
+		dsdEndpoint:      os.Getenv(internal.DatadogDSDEndpoint),
+		metricSampleRate: defaultMetricSampleRate,
+		tags: []string{
+			fmt.Sprintf("environment:%s", os.Getenv(internal.DatadogEnvironment)),
+			fmt.Sprintf("service:%s", os.Getenv(internal.DatadogService)),
+			fmt.Sprintf("version:%s", os.Getenv(internal.DatadogVersion)),
+		},
+	}
+
+	for _, option := range opts {
+		err = option(options)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return options, nil
+}
+
+// WithTags sets the tags that are sent with every metric, shorthand for
+// statsd.WithTags()
+func WithTags(tags ...string) Option {
+	return func(options *options) error {
+		options.tags = append(options.tags, tags...)
```

Comment:
What's the default behaviour of underlying library if the user of this library tries to set those default tags?

If the intention is to not allow setting them, should be not add some sanity checks to block users from doing so?

Reply:
Late to the party here sorry. From what I've seen of the datadog metric lib, I think the absolute minimum we need is env & service. Version is probably sort of required. 

I'd go as far as saying that to start the whole datadog stack, you must send these three values. 
```go
stop := coopdatadog.Start(ctx, "production", "my-service", "7974730")
defer stop()
```

Reply:
If we enforce that we cannot start without these three values then we can safely ignore any extra tags with the `env`, `service` and `version` names.

Reply:
environment, service name and and version are picked up from required environmental variables.

https://github.com/coopnorge/go-datadog-lib/pull/434/files#diff-ec6d18a673de2007558daf63e89887c95624876f2b293e41b232ca8c1b14472eR27-R58

Reply:
Reopening, sorry I looked at this before my morning coffee!

Reply:
A tag in Datadog has do parts, a key and a value on the form `<key>:<value>`. I'm not sure it is worth spending time on trying to limit duplicates etc until we see this in the wild.